runtime/parachains/src/configuration/migration/v7.rs

```diff
@@ -0,0 +1,315 @@
+		log::trace!(target: crate::configuration::LOG_TARGET, "Running pre_upgrade() for MigrateToV7");
+		Ok(Vec::new())
+	}
+
+	fn on_runtime_upgrade() -> Weight {
+		log::info!(target: configuration::LOG_TARGET, "MigrateToV7 started");
+		if StorageVersion::get::<Pallet<T>>() == 6 {
+			let weight_consumed = migrate_to_v7::<T>();
+
+			log::info!(target: configuration::LOG_TARGET, "MigrateToV7 executed successfully");
+			StorageVersion::new(7).put::<Pallet<T>>();
+
+			weight_consumed
+		} else {
+			log::warn!(target: configuration::LOG_TARGET, "MigrateToV7 should be removed.");
+			T::DbWeight::get().reads(1)
+		}
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(_state: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+		log::trace!(target: crate::configuration::LOG_TARGET, "Running post_upgrade()");
+		ensure!(
+			StorageVersion::get::<Pallet<T>>() >= 7,
+			"Storage version should be >= 7 after the migration"
+		);
+
+		Ok(())
+	}
+}
+
+fn migrate_to_v7<T: Config>() -> Weight {
+	// Unusual formatting is justified:
+	// - make it easier to verify that fields assign what they supposed to assign.
+	// - this code is transient and will be removed after all migrations are done.
+	// - this code is important enough to optimize for legibility sacrificing consistency.
+	#[rustfmt::skip]
+	let translate =
+		|pre: V6HostConfiguration<BlockNumberFor<T>>| ->
+		V7HostConfiguration<BlockNumberFor<T>>
+	{
+		V7HostConfiguration {
+max_code_size                            : pre.max_code_size,
+max_head_data_size                       : pre.max_head_data_size,
+max_upward_queue_count                   : pre.max_upward_queue_count,
+max_upward_queue_size                    : pre.max_upward_queue_size,
+max_upward_message_size                  : pre.max_upward_message_size,
+max_upward_message_num_per_candidate     : pre.max_upward_message_num_per_candidate,
+hrmp_max_message_num_per_candidate       : pre.hrmp_max_message_num_per_candidate,
+validation_upgrade_cooldown              : pre.validation_upgrade_cooldown,
+validation_upgrade_delay                 : pre.validation_upgrade_delay,
+max_pov_size                             : pre.max_pov_size,
+max_downward_message_size                : pre.max_downward_message_size,
+hrmp_max_paras_outbound_channels     : pre.hrmp_max_parachain_outbound_channels,
+hrmp_sender_deposit                      : pre.hrmp_sender_deposit,
+hrmp_recipient_deposit                   : pre.hrmp_recipient_deposit,
+hrmp_channel_max_capacity                : pre.hrmp_channel_max_capacity,
+hrmp_channel_max_total_size              : pre.hrmp_channel_max_total_size,
+hrmp_max_paras_inbound_channels      : pre.hrmp_max_parachain_inbound_channels,
+hrmp_channel_max_message_size            : pre.hrmp_channel_max_message_size,
+code_retention_period                    : pre.code_retention_period,
+parathread_cores                         : pre.parathread_cores,
```

Comment:
Shouldn't we rename these to on_demand as well?

Reply:
Hmm. No, that's not what I meant. I believe Alex introduced `paras` to refer to all kinds of parachains, which is fine I guess. What I meant is the entries that particularly still refer to parathreads explicitly. Since we renamed them to on-demand, there should be no `parathread_....` options either.

Reply:
Oh yes, got it.